### PR TITLE
HDFS-17310. DiskBalancer: Enhance the log message for submitPlan

### PR DESCRIPTION
**Pull request by haiyang1987**
### Description of PR
https://issues.apache.org/jira/browse/HDFS-17310

In order to convenient troubleshoot problems, enhance the log message for submitPlan.

### CODE REVIEW

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DiskBalancer.java [hunk 1]
```diff
@@ -443,7 +444,7 @@ private NodePlan verifyPlanHash(String planID, String plan)
       throws DiskBalancerException {
     final long sha1Length = 40;
     if (plan == null || plan.length() == 0) {
-      LOG.error("Disk Balancer -  Invalid plan.");
+      LOG.error("Disk Balancer -  Invalid plan ().");
```
Comment (slfan1989) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DiskBalancer.java]: This seems to be missing some information.

Comment (haiyang1987) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DiskBalancer.java]: Thanks @slfan1989 for your comment.
here no need to modify it，i will remove ‘()’

**Review by slfan1989 — APPROVED** [review]

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DiskBalancer.java [hunk 2]
```diff
@@ -523,7 +524,7 @@ private void createWorkPlan(NodePlan plan) throws DiskBalancerException {
 
       String sourceVolBasePath = storageIDToVolBasePathMap.get(sourceVolUuid);
       if (sourceVolBasePath == null) {
-        final String errMsg = "Disk Balancer - Unable to find volume: "
+        final String errMsg = "Disk Balancer - Unable to find source volume: "
```
Comment (slfan1989) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DiskBalancer.java:527]: Can we use `{}` instead of `+` ?

**Comment by haiyang1987** [comment]
Hi @tasanuma Can you help to take a review this small changes when you have time? thank you very much~

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  12m  6s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  42m 33s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m 19s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   1m 13s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  checkstyle  |   1m  7s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 23s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m  4s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 38s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  spotbugs  |   3m 10s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  34m 24s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 10s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 12s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   1m 12s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  5s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |   1m  5s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 56s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 13s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 52s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 29s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  spotbugs  |   3m 11s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  34m 12s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 208m 41s |  |  hadoop-hdfs in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 42s |  |  The patch does not generate ASF License warnings.  |
|  |   | 354m 57s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6391/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6391 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 00d50bf38832 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / dfc915e94292755877743d38975e5189f6ec68da |
| Default Java | Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6391/1/testReport/ |
| Max. process+thread count | 3645 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6391/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by hotcodemacha — APPROVED** [review]

**Comment by haiyang1987** [comment]
Thanks @slfan1989 @ashutoshcipher help me reivew it. 
Could you mind to push this modification forward when you have free time ? Thank you very much.

**Comment by slfan1989** [comment]
> Thanks @slfan1989 @ashutoshcipher help me reivew it.
> Could you mind to push this modification forward when you have free time ? Thank you very much.

@haiyang1987  Thanks for the contribution! LGTM. 

But do we wait 1-2 working days for tasanuma to help review the PR?

cc:@ashutoshcipher

**Review by tasanuma — APPROVED** [review]
Sorry for being late.
LGTM. +1.

**Comment by slfan1989** [comment]
@haiyang1987 Thank you for your contribution! Merged Into trunk. @ashutoshcipher @tasanuma Thanks for reviewing the code!

**Comment by haiyang1987** [comment]
Thanks @slfan1989 @tasanuma @ashutoshcipher for your review and merge it!

**Pull request merged**